app/models/group.rb

```diff
@@ -9,7 +12,9 @@ class Group < ActiveRecord::Base
   scope :students, -> { where(name: 'Students') }
   scope :coaches, -> { where(name: 'Coaches') }
 
-  validates :name, :chapter_id, presence: true
+  validates :name, presence: true
+
+  validate :valid_name?
```

Comment:
You could save yourself a bit of code here. Rails has a built-in validator that does exactly this, it works like this:

```ruby
validates :name, presence: true, 
                 inclusion: {in: NAMES, message: "Invalid name for Group"}
```

One minor difference is that the above does add the error to the `name` field rather than the `group` field like your `valid_name?` does. But that might actually be what you wanted in the first place, since that's what the HTML form is using too (and `.group` doesn't actually seem to be a method here anyways?)